src/System.Net.Http/src/System/Net/Http/HttpClientHandler.Core.cs

```diff
@@ -46,6 +49,45 @@ public long MaxRequestContentBufferSize
             }
         }
 
+        protected internal override Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
+        {
+            if (request == null)
```

Comment:
@scalablecory 
I know we've gone back and forth on this, but this particular null check _must_ be performed because it is called from https://github.com/dotnet/corefx/blob/2fc48e13e87a4e68576d7df7af420ac2874e929e/src/System.Net.Http/tests/FunctionalTests/DiagnosticsTests.cs#L1166-L1203 .

The specific call chain is MyHandler.SendAsync(HttpRequestMessage) => HttpClientHandler.SendAsync(HttpRequestMessage, CancellationToken) => ... => DiagnosticsHandler.SendAsync(HttpRequestMessage, CancellationToken).

The other redundant null checks have been removed.

Reply:
HttpClient is not the only class that can call the protected internal `SendAsync` method of a handler.  `HttpMessageInvoker` is the base class of HttpClient and it can call the `SendAsync` method of a handler. Generally speaking, we should leave in the check for `null` request message.